vault/resource_pki_secret_backend_crl_config.go

```diff
@@ -0,0 +1,118 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func pkiSecretBackendCrlConfigResource() *schema.Resource {
+	return &schema.Resource{
+		Create: pkiSecretBackendCrlConfigCreate,
+		Read:   pkiSecretBackendCrlConfigRead,
+		Update: pkiSecretBackendCrlConfigUpdate,
+		Delete: pkiSecretBackendCrlConfigDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The path of the PKI secret backend the resource belongs to.",
+			},
+			"expiry": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the time until expiration.",
+			},
+			"disable": {
+				Type:        schema.TypeBool,
+				Required:    false,
+				Optional:    true,
+				Description: "Disables or enables CRL building",
+			},
+		},
+	}
+}
+
+func pkiSecretBackendCrlConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+
+	path := pkiSecretBackendCrlConfigPath(backend)
+
+	expiry := d.Get("expiry")
+	disable := d.Get("disable")
```

Comment:
For these two, since they're optional, I think we might want to do something more like this instead:
```
data := make(map[string]interface{})
if expiry, ok := d.GetOK("expiry"); ok {
    data["expiry"] = expiry
}
if disable, ok := d.GetOK("disable"); ok {
    data["disable"] = disable
}
```
Then they're only sent to Vault if they were actually specified by the user. This will allow Vault to see that they're unset and use its normal defaults.

Reply:
I have taken on board you suggestion and incorporated this in both the Create and Update functions.